app/src/main/java/com/example/myapplication/vm/LoginViewModel.kt

```diff
@@ -50,22 +57,53 @@ class LoginViewModel(app: Application) : AndroidViewModel(app) {
         _emptyFieldsError.value = null
     }
 
-     fun getUserInfo(userAuth: UserAuth) {
-         viewModelScope.launch {
-             val response = userRepository.getUserInfo(userAuth)
-             if(response.isSuccessful){
-                 val editor = sharedPreferences.edit()
-                 editor.also {
-                     it.putString(USER_INFO, Gson().toJson(response.body()))
-                     it.putString(USERNAME, userAuth.email)
-                     it.putString(PASSWORD, userAuth.password)
-                     it.commit()
-                 }
-                 _userResponseIsSuccessful.value = true
-             }else{
-                 _userResponseIsSuccessful.value = false
-             }
-         }
+    fun statRequest(userAuth: UserAuth) {
+        val deviceOnline = isOnline(getApplication())
+        if (deviceOnline) {
+            getUserInfo(userAuth)
+        } else {
+            _userResponse.value = NetworkResponseState.NO_INTERNET_CONNECTION
+        }
+    }
+
+    private fun isOnline(context: Context): Boolean {
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val capabilities =
+            connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+        if (capabilities != null) {
+            if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
```

Comment:
You can have an (OR) here:
return capabilities.has(A) || capabilities.has(B) || capabilities.has(C).